activerecord/test/cases/calculations_test.rb

```diff
@@ -460,6 +460,44 @@ def test_should_count_manual_select_with_include
     assert_equal 6, Account.select("DISTINCT accounts.id").includes(:firm).count
   end
 
+  def test_should_count_manual_select_with_count_all
+    assert_equal 5, Account.select("DISTINCT accounts.firm_id").count(:all)
+  end
+
+  def test_should_count_with_manual_distinct_select_and_distinct
+    assert_equal 4, Account.select("DISTINCT accounts.firm_id").distinct(:id).count
+  end
+
+  def test_should_count_manual_select_with_group_with_count_all
+    c = Account.select("DISTINCT accounts.firm_id").group("accounts.firm_id").count(:all)
+    assert_equal([[nil, 1], [1, 1], [2, 1], [6, 2], [9, 1]].to_h, c)
+  end
+
+  def test_should_count_manual_with_count_all
+    assert_equal 6, Account.count(:all)
+  end
+
+  def test_size_manual_select_distinct
+    assert_equal 5, Account.select("DISTINCT accounts.firm_id").size
+  end
+
+  def test_size_with_manual_distinct_select_with_loaded_size
+    size = Account.select("DISTINCT accounts.firm_id").size
+    loaded_size = Account.select("DISTINCT accounts.firm_id").load.size
+    assert_equal size, loaded_size
```

Comment:
> Also I think the test cases about `size` shouldn't be on `test/cases/calculations_test.rb` file

Yes, just put them in `test/cases/relations_test.rb` for now.

https://github.com/rails/rails/blob/be442ccd8342baf39e73b01a6fdd38e1a3b15c2c/activerecord/test/cases/relations_test.rb#L957-L973

Reply:
There is a weird situation added a test with `includes`:
```ruby
def test_size_with_eager_loading_and_manual_distinct_select_and_includes
  accounts = Account.select("DISTINCT accounts.firm_id").includes(:firm).order("firms.id")

  assert_queries(1) { assert_equal 5, accounts.size }
  assert_queries(1) { assert_equal 5, accounts.load.size }
end
```
It raises:
`# ActiveRecord::StatementInvalid: SQLite3::SQLException: near "DISTINCT": syntax error`
I can investigate further if you want. 

But this also gets raised when using:
```ruby
Account.distinct(:credit_limit).select("DISTINCT accounts.firm_id").count(:all)
# ActiveRecord::StatementInvalid: SQLite3::SQLException: near "DISTINCT": syntax error
```

What do you think about this?

Reply:
This is an issue for `count` with eager loading.
I'd prefer to address it in another PR, since it is a little complex than what this PR fixes.

Reply:
Ok I'll work it in another PR!